### PR TITLE
Create cisco_ios_show_interfaces_description.template

### PR DESCRIPTION
**Pull request by adrydale**
##### ISSUE TYPE
 - New Template Pull Request

##### COMPONENT
Add new template for "show interfaces description" on Cisco IOS devices

##### SUMMARY
No template for "show interfaces description" currently exists.

```
tulqtcdskd4s04p#show interfaces description
Interface                      Status         Protocol Description
Vl1                            admin down     down     
Vl99                           up             up       10.20.99.0_Switch_mgmt_VLAN
Gi0/1                          down           down     D3 USER
Gi0/2                          down           down     D3 USER
Gi0/3                          down           down     D3 USER
Gi0/4                          down           down     D3 USER
Gi0/5                          down           down     D3 USER
Gi0/6                          down           down     D3 USER
Gi0/7                          down           down     D3 USER
Gi0/8                          up             up       MERAKI TEST AP
Gi0/9                          admin down     down     
Gi0/10                         up             up       UPLINK TO TULCCD3S01P
tulqtcdskd4s04p#
```

```
>>> print(yaml.dump(parsed_data,explicit_start=True, default_flow_style=False))
- - Vl1
  - admin down
  - down
  - ' '
- - Vl99
  - up
  - up
  - 10.20.99.0_Switch_mgmt_VLAN
- - Gi0/1
  - down
  - down
  - D3 USER
- - Gi0/2
  - down
  - down
  - D3 USER
- - Gi0/3
  - down
  - down
  - D3 USER
- - Gi0/4
  - down
  - down
  - D3 USER
- - Gi0/5
  - down
  - down
  - D3 USER
- - Gi0/6
  - down
  - down
  - D3 USER
- - Gi0/7
  - down
  - down
  - D3 USER
- - Gi0/8
  - up
  - up
  - MERAKI TEST AP
- - Gi0/9
  - admin down
  - down
  - ' '
- - Gi0/10
  - up
  - up
  - UPLINK TO TULCCD3S01P
```

### CODE REVIEW

**Comment by adrydale** [comment]
@jmcgill298, does this pull look good? I closed the previous pull and re-did it with the test files.

File: templates/cisco_ios_show_interfaces_description.template [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Value PORT (\S+)
+Value STATUS (up|down|admin down)
+Value PROTOCOL (up|down)
+Value DESCRIP (.+?)
+
+Start
+  ^Interface -> Begin
```
Comment (jmcgill298) [inline templates/cisco_ios_show_interfaces_description.template]:
```suggestion
  ^Interface\s+Status\s+Protocol\s+Description\s*$$ -> Begin
  ^\s*$$
  ^. -> Error
```
The header column will verify that the template is capturing the correct data. Another version of the OS or a different platform will have different headers, and then the template will inappropriately record the data.

File: templates/cisco_ios_show_interfaces_description.template [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+Begin
+  ^${PORT}\s+${STATUS}\s+${PROTOCOL}\s+${DESCRIP}$$ -> Record
+  ^.*$$ -> Error
```
Comment (jmcgill298) [inline templates/cisco_ios_show_interfaces_description.template]:
```suggestion
  ^\s*$$
  ^. -> Error
```

File: templates/cisco_ios_show_interfaces_description.template [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Value PORT (\S+)
+Value STATUS (up|down|admin down)
+Value PROTOCOL (up|down)
+Value DESCRIP (.+?)
+
+Start
+  ^Interface -> Begin
+
+Begin
+  ^${PORT}\s+${STATUS}\s+${PROTOCOL}\s+${DESCRIP}$$ -> Record
```
Comment (jmcgill298) [inline templates/cisco_ios_show_interfaces_description.template]:
```suggestion
  ^${PORT}\s+${STATUS}\s+${PROTOCOL}(?:\s+${DESCRIP})?\s*$$ -> Record
```
Allows for a trailing space
Allows for `DESCRIP` to be empty and not record a space

File: templates/cisco_ios_show_interfaces_description.template [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Value PORT (\S+)
+Value STATUS (up|down|admin down)
```
Comment (jmcgill298) [inline templates/cisco_ios_show_interfaces_description.template]:
```suggestion
Value STATUS (up|down|admin\s+down)
```

File: templates/cisco_ios_show_interfaces_description.template [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+Value PROTOCOL (up|down)
+Value DESCRIP (.+?)
```
Comment (jmcgill298) [inline templates/cisco_ios_show_interfaces_description.template]:
```suggestion
Value DESCRIP (.*?)
```
Allow for `DESCRIP` to be  empty and not record a space

File: tests/cisco_ios/show_interfaces_description/cisco_ios_show_interfaces_description.parsed [hunk 6]
```diff
@@ -0,0 +1,51 @@
+---
+parsed_sample:
+
+- port: Vl1
+  status: admin down
+  protocol: down
+  descrip: ' '
```
Comment (jmcgill298) [inline tests/cisco_ios/show_interfaces_description/cisco_ios_show_interfaces_description.parsed]:
```suggestion
  descrip: ''
```

File: tests/cisco_ios/show_interfaces_description/cisco_ios_show_interfaces_description.parsed [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  protocol: down
+  descrip: D3 USER
+- port: Gi0/8
+  status: up
+  protocol: up
+  descrip: MERAKI TEST AP
+- port: Gi0/9
+  status: admin down
+  protocol: down
+  descrip: ' '
```
Comment (jmcgill298) [inline tests/cisco_ios/show_interfaces_description/cisco_ios_show_interfaces_description.parsed]:
```suggestion
  descrip: ''
```

**Comment by adrydale** [comment]
Thank you for the changes, @jmcgill298! I've added them and updated the index file. I tested the template again on my devices and it's working well.

**Comment by jmcgill298** [comment]
Thanks @adrydale the index ordering was slightly off, so I fixed that and will merge.

**Pull request merged**